src/librustc_typeck/astconv.rs

```diff
@@ -1731,8 +1731,14 @@ pub fn partition_bounds<'a>(tcx: &ty::ctxt,
                         if ty::try_add_builtin_trait(tcx,
                                                      trait_did,
                                                      &mut builtin_bounds) {
-                            // FIXME(#20302) -- we should check for things like Copy<T>
-                            continue; // success
+                            let segments = &b.trait_ref.path.segments;
+                            let parameters = &segments[segments.len() - 1].parameters;
+                            if parameters.is_empty() {
+                                continue; // success
+                            }
+                            span_fatal!(tcx.sess, b.trait_ref.path.span, E0316,
```

Comment:
In general, every span_fatal is a bug imo, though sometimes there are good reasons. In this particular case, I think those other errors are holdovers and could be made span_err with some effort. (span_err can be trickier because you have to have some kind of intelligent fallback behavior...)